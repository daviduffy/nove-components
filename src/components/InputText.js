import React from 'react';
import PropTypes from 'prop-types';

const InputText = ({
  id,
  label,
  placeholder,
  classes,
  error,
  value,
  options,
  onChange,
  style,
  valueRequired,
  ...rest
}) => (
  <div className={`F__g${classes ? ` ${classes}` : ''}`} style={style}>
    {
      label &&
      <label htmlFor={id} className={`F__l${valueRequired ? ' F__l--req' : ''}`}>
        <span>{label}</span>
      </label>
    }
    <input
      id={id}
      type={id === 'email' ? 'email' : 'text'}
      className={`F__i${error ? ' F__i--err' : ''}`}
      onChange={onChange}
      placeholder={placeholder}
      value={value}
      {...rest}
    />
  </div>
);
InputText.propTypes = {
  classes: PropTypes.string,
  error: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.bool,
  placeholder: PropTypes.string,
  style: PropTypes.object,
  value: PropTypes.string.isRequired,
  valueRequired: PropTypes.bool
};
InputText.defaultProps = {
  label: '',
  classes: '',
  options: false,
  placeholder: '',
  style: {},
  valueRequired: false
};

export default InputText;
