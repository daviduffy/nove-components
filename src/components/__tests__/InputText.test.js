// External Dependencies
import React from 'react';
import { shallow } from 'enzyme';

// Internal Dependencies
import InputText from '../InputText';
import { text as textProps } from '../../fixtures/input';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<InputText {...textProps} />);
});

test('should render InputText correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should pass in custom classes if present', () => {
  const classes = 'bubberge';
  expect(wrapper.find(`.${classes}`)).toHaveLength(0);
  const errorProps = {
    ...textProps,
    classes
  };
  wrapper = shallow(<InputText {...errorProps} />);
  expect(wrapper.find(`.${classes}`)).toHaveLength(1);
});

test('should render with error class when error is present', () => {
  expect(wrapper.find('.F__i--err')).toHaveLength(0);
  const errorProps = {
    ...textProps,
    error: true
  };
  wrapper = shallow(<InputText {...errorProps} />);
  expect(wrapper.find('.F__i--err')).toHaveLength(1);
});

test('should render with required class when required', () => {
  expect(wrapper.find('.F__l--req')).toHaveLength(0);
  const requiredProps = {
    ...textProps,
    required: true
  };
  wrapper = shallow(<InputText {...requiredProps} />);
  expect(wrapper.find('.F__l--req')).toHaveLength(1);
});

test('should change to a different input type if type supplied', () => {
  expect(wrapper.find('[type="text"]')).toHaveLength(1);
  expect(wrapper.find('[type="number"]')).toHaveLength(0);
  const numberProps = {
    ...textProps,
    type: 'number'
  };
  wrapper = shallow(<InputText {...numberProps} />);
  expect(wrapper.find('[type="text"]')).toHaveLength(0);
  expect(wrapper.find('[type="number"]')).toHaveLength(1);
});
