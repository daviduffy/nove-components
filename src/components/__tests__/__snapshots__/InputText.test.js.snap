// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputText correctly 1`] = `
<div
  className="F__g F__g--name"
  style={Object {}}
>
  <label
    className="F__l"
    htmlFor="name"
  >
    <span>
      Input Label
    </span>
  </label>
  <input
    autoFocus={false}
    className="F__i"
    id="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value="Bim Ellen"
  />
</div>
`;
